operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractNonNamespacedResourceOperatorTest.java

```diff
@@ -465,12 +466,57 @@ public void testReconcileDeletionThrowsWhenDeleteMethodReturnsFalse(VertxTestCon
         AbstractNonNamespacedResourceOperator<C, T, L, R> op = createResourceOperations(vertx, mockClient);
 
         Checkpoint async = context.checkpoint();
-        op.reconcile(Reconciliation.DUMMY_RECONCILIATION, resource.getMetadata().getName(), null).onComplete(context.failing(e -> {
+        op.reconcile(Reconciliation.DUMMY_RECONCILIATION, resource.getMetadata().getName(), null).onComplete(context.succeeding(rrDeleted -> {
             verify(mockResource).delete();
             context.verify(() -> assertThat("Watch was not closed", watchWasClosed.get(), is(true)));
             async.flag();
         }));
     }
 
+    // This tests the pre-check which should stop the self-closing-watch in case the resource is deleted before the
+    // watch is opened.
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testReconcileDeleteDoesNotTimeoutWhenResourceIsAlreadyDeleted(VertxTestContext context) {
+        T resource = resource();
+        AtomicBoolean watchWasClosed = new AtomicBoolean(false);
+        Resource mockResource = mock(resourceType());
+
+        AtomicInteger getInvocations = new AtomicInteger(0);
```

Comment:
Rather than counting invocations why not create this mock after the mock for `watch()` and have the return from this method predicated on whether the watch was created?

Reply:
Please have a look if the change I made is what you meant.